### PR TITLE
Simplify pos_is_ok()

### PR DESCRIPTION
**Pull request by mcostalba**
Now we don't need anymore the tricky pointer to
show the failed test.

Also small rename in see_ge() while there.

No functional change.

### CODE REVIEW

File: src/position.cpp [hunk 1]
```diff
@@ -1146,78 +1146,65 @@ void Position::flip() {
...
-              ||(pieces(WHITE) | pieces(BLACK)) != pieces())
-              return false;
 
-          for (PieceType p1 = PAWN; p1 <= KING; ++p1)
-              for (PieceType p2 = PAWN; p2 <= KING; ++p2)
-                  if (p1 != p2 && (pieces(p1) & pieces(p2)))
-                      return false;
-      }
+  if (  (pieces(WHITE) & pieces(BLACK))
+      ||(pieces(WHITE) | pieces(BLACK)) != pieces())
```
Comment (Rocky640) [inline src/position.cpp]: missing a space after the ||

File: src/position.cpp [hunk 2]
```diff
@@ -1146,78 +1146,65 @@ void Position::flip() {
...
 
+  for (Color c = WHITE; c <= BLACK; ++c)
+      for (CastlingSide s = KING_SIDE; s <= QUEEN_SIDE; s = CastlingSide(s + 1))
+      {
+          if (!can_castle(c | s))
+              continue;
+
+          if (   piece_on(castlingRookSquare[c | s]) != make_piece(c, ROOK)
+              || castlingRightsMask[castlingRookSquare[c | s]] != (c | s)
+              ||(castlingRightsMask[square<KING>(c)] & (c | s)) != (c | s))
```
Comment (Rocky640) [inline src/position.cpp]: Missing a space after the ||

Comment (Rocky640) [inline src/position.cpp]: In pos_is_ok, would it make sense to add code to
a) check total count <=16 for each side
b) no pawns on rank 1 or 8
c) extra minors or majors + pawn count <=8

**Comment by mcostalba** [comment]
@Rocky640 thanks. Updated.

I would not add more tests to avoid adding too much complexity, this is meant to remain simple.

File: src/position.cpp [hunk 3]
```diff
@@ -1146,78 +1145,73 @@ void Position::flip() {
...
-          if (   std::count(board, board + SQUARE_NB, W_KING) != 1
-              || std::count(board, board + SQUARE_NB, B_KING) != 1
-              || attackers_to(square<KING>(~sideToMove)) & pieces(sideToMove))
-              return false;
+  if (   pieceCount[W_KING] != 1
+      || pieceCount[B_KING] != 1
+      || attackers_to(square<KING>(~sideToMove)) & pieces(sideToMove))
+      assert(0 && "pos_is_ok: Kings");
+
+  if (   (pieces(WHITE, PAWN) & Rank1BB)
```
Comment (Rocky640) [inline src/position.cpp]: Suggestion: replace line 1171 and 1172 with 
`if ((pieces(PAWN) & (Rank1BB | Rank8BB))`

**Comment by snicolet** [comment]
@mcostalba 
In Position::set(), you are now using rows 2 and 7 to put the pieces and the pawns. By looking at the code, it seems that a `/` is missing before the last row in line 397 of position.cpp (there used to be seven `/` separators, now I count only six).

**Comment by mcostalba** [comment]
@Rocky640 @snicolet thanks! updated.

**Comment by mstembera** [comment]
I noticed we use a lot of
```C++
assert(0 && "Some string");
```
A small syntax trick if u like it is to just write
```C++
assert(!"Some string");
```

**Pull request closed without merging**